src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,815 @@
+            # will be incorrectly matched together.
+            # TODO Handle the corner case
+            fib_fiducials = []
+            fm_fiducials = []
+            self.correlation_target.fm_pois = []
+            for target in self._tab_data_model.main.targets.value:
+                if "FIB" in target.name.value:
+                    fib_fiducials.append(target)
+                elif "FM" in target.name.value:
+                    fm_fiducials.append(target)
+                elif "POI" in target.name.value:
+                    self.correlation_target.fm_pois = target
+            if fib_fiducials:
+                fib_fiducials.sort(key=lambda x: x.index.value)
+                self.correlation_target.fib_fiducials = fib_fiducials
+            if fm_fiducials:
+                fm_fiducials.sort(key=lambda x: x.index.value)
+                self.correlation_target.fm_fiducials = fm_fiducials
+
+        acq_conf = conf.get_acqui_conf()
+        save_features(acq_conf.pj_last_path, self._tab_data_model.main.features.value)
+
+    def check_correlation_conditions(self) -> bool:
+        """
+        Minimum 4 FIB and FM fiducials and 1 POI in FM are required to run the correlation.
+        :return: (bool) True if the conditions are met, False otherwise
+        """
+        if self.correlation_target:
+            if (len(self.correlation_target.fib_fiducials) >= 4 and len(
+                    self.correlation_target.fm_fiducials) >= 4 and self.correlation_target.fm_pois and len(
+                self._tab_data_model.views.value[0].stream_tree) > 0) and self.correlation_target.fib_stream:
+                return True
+            else:
+                self.correlation_target.reset_attributes()
+                self._tab_data_model.projected_points = []
+                self.correlation_txt.SetLabel("Correlation RMS Deviation :")
+                for vp in self._viewports:
+                    if vp.view.name.value == "SEM Overview":
+                        vp.canvas.update_drawing()
+                return False
+        else:
+            return False
+
+    def _queue_latest_change(self):
+        """
+        This method is called when there's an input change.
+        If there's a change already being processed, it will override it and process the latest change.
+        """
+        with self.lock:  # Ensure thread-safe access to latest_change
+            self.latest_change = True  # Overwrite with the latest change
+            if self.is_processing:  # If a process is running, it will handle the latest change.
+                logging.warning("Multipoint Correlation is running. It will handle the latest change.")
+                self.correlation_txt.SetLabel("Correlation RMS Deviation :  Calculating...")
+                return
+            # Start processing the latest change in a separate thread
+            self.process_thread = threading.Thread(target=self._process_latest_change)
+            self.process_thread.start()
```

Comment:
This is quite complex threading code. Typically, a simple way is to do the following:
* start a thread at init (which runs the whole time in the background, as "daemon")
* this thread waits for data coming from queue.Queue
* whenever new processing is required, it is placed on the queue.
* to make sure it only processes the latest request, after getting a request, it can check for more data on the queue, non-blocking (or just repeat until the queue is empty)
* to request the thread to stop, pass a special data, for instance `None`.

=> You don't need locks. You only start the thread once. You are certain there is only one computation running at a given time.